File: LeetcodeTasks/MissingNumber/MissingNumberTests.cs
using NUnit.Framework;
using NUnit.Framework.Constraints;

namespace LeetcodeTasks.MissingNumber
{
	public class MissingNumberTests
	{
		[Test]
		public void NullOremptyArray_Success()
		{
			int[] arr = null;
			var sol = new MissingNumber(arr);
			var result = sol.GetMissingNumner();

			Assert.AreEqual(-1, result);
		}

		[Test]
		public void FirstArray_Success()
		{
			int[] arr = { 3,0,1};
			var sol = new MissingNumber(arr);
			var result = sol.GetMissingNumner();

			Assert.AreEqual(2, result);
		}


		[Test]
		public void SecondArray_Success()
		{
			int[] arr = { 9, 6, 4, 2, 3, 5, 7, 0, 1 };
			var sol = new MissingNumber(arr);
			var result = sol.GetMissingNumner();

			Assert.AreEqual(8, result);
		}

		//============================================================

		[Test]
		public void SecondArray_SecondApproach_Success()
		{
			int[] arr = { 9, 6, 4, 2, 3, 5, 7, 0, 1 };
			var sol = new MissingNumber(arr);
			var result = sol.GetMissingNumberBits();

			Assert.AreEqual(8, result);
		}
	}
}

File: GrokkingCoding/src/com/fruitfactory/cyclicsort/missingnumber/MissingNumberTests.java
package com.fruitfactory.cyclicsort.missingnumber;

import org.junit.Test;
import static org.junit.Assert.*;

public class MissingNumberTests {

    @Test
    public void FirstTest(){
        int[] arr = new int[]{4, 0, 3, 1};
        MissingNumber mn = new MissingNumber();
        int result = mn.findMissingNumber(arr);
        assertEquals(2, result);
    }

    @Test
    public void SecondTest(){
        int[] arr = new int[]{8, 3, 5, 2, 4, 6, 0, 1};
        MissingNumber mn = new MissingNumber();
        int result = mn.findMissingNumber(arr);
        assertEquals(7, result);
    }

}
